app/src/main/java/edu/uw/ischool/opensecrets/SecretApp.kt

```diff
@@ -1,17 +1,171 @@
 package edu.uw.ischool.opensecrets
 
+import android.Manifest
+import android.app.AlarmManager
 import android.app.Application
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.widget.Toast
+import edu.uw.ischool.opensecrets.model.ContactMessager
+import edu.uw.ischool.opensecrets.model.Entry
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+import java.io.File
+import java.io.FileReader
+import java.io.FileWriter
+import android.content.Context
+import android.content.IntentFilter
+import android.content.pm.PackageManager
+import android.os.Build
+import android.util.Log
+import androidx.core.app.ActivityCompat
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import edu.uw.ischool.opensecrets.model.Contact
+import edu.uw.ischool.opensecrets.model.TimedMessageQueue
+import java.util.Date
 import edu.uw.ischool.opensecrets.util.JournalManager
 import edu.uw.ischool.opensecrets.util.OptionManager
 
 
+const val CHANNEL_ID = "OpenSecretNotifications"
 class SecretApp : Application() {
+
+    private lateinit var journal: File
+    private val contactMessager : ContactMessager = ContactMessager()
+    lateinit var messageQueue : TimedMessageQueue
+    lateinit var alarmManager: AlarmManager
+    private var messageID = 1
+    private val sampleEntry : Entry = Entry("My Title", "blue", "Call me Ishmael. Some years ago...", Date())
     lateinit var journalManager: JournalManager
     lateinit var optionManager: OptionManager
     override fun onCreate() {
         journalManager = JournalManager(this)
         optionManager = OptionManager(this)
         super.onCreate()
+        createNotificationChannel()
+        alarmManager = getSystemService(Context.ALARM_SERVICE) as AlarmManager
+        messageQueue = TimedMessageQueue(this, alarmManager, ::sendEntryToRandomContact)
+        updateContactList()
+//        sendNotification("Notification title", "Hello there!")
+//        queueSendEntry(sampleEntry)
+    }
+
+    
+
+    fun journalExist(): Boolean {
```

Comment:
remove this since JournalManager has been implemented, so call JournalManager instead for this.